moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -329,8 +328,26 @@ void ompl_interface::ModelBasedPlanningContext::interpolateSolution()
   if (ompl_simple_setup_->haveSolutionPath())
   {
     og::PathGeometric& pg = ompl_simple_setup_->getSolutionPath();
-    pg.interpolate(
```

Comment:
According to the documentation of `og::PathGeometric::interpolate(unsigned int count)`, this change is not necessary:
> Changes are performed only if a path has less than count states.

Reply:
> However, this re-interpolates as well and thus again might cause the issue addressed in this PR.

Yep, that's why this change was made. Originally, we just had
```
pg.interpolate();
if (pg.getStateCount() < minimum_waypoint_count_)
  pg.interpolate(minimum_waypoint_count_);
```
But we realized that trying to add just a few evenly interpolated states to an already evenly interpolated path without changing any of the original states would result in several parts of the paths having more states than the rest of the path, resulting in jerky trajectories.

Reply:
I don't get why do you re-interpolate at all? According to your initial comment, this is not a good idea.
Probably I didn't understand the semantics of pg.interpolate() yet. Is it performing the same interpolation as during planning, because these sub waypoints are not actually stored in the path anymore? Then, I see your point.